servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java

```diff
@@ -488,4 +491,14 @@ private static int verifyPortRange(final int port) {
         }
         return port;
     }
+
+    private static void setHostHeader(HttpRequestMetaData metaData) {
+        if (!HTTP_1_0.equals(metaData.version()) && !metaData.headers().contains(HOST)) {
```

Comment:
Alternatively, we can use a pattern similar to `HeaderUtils` and others places, where we do something like: `metaData.version().major() >= 1 && metaData.version().minor() >= 1`. It makes it clear that it's required only for 1.1 and above.

Reply:
Is that meaningfully different from what I have? Your code snippet doesn't seem to work for HTTP/2. We could use the `compare(..)` method but that always feels harder to read.

Reply:
🤦 yes, I made a mistake with h2
np, we can keep as-is